lib/stages/visAccumulate.cpp

```diff
@@ -361,19 +344,27 @@ void visAccumulate::main_thread() {
         // If we've got to here and we've not initialised we need to skip this frame.
         if (init) {
 
-            // Now the main accumulation work starts...
+            // Get the amount of data in the frame
+            // TODO: for the multifrequency support this probably needs to become frequency
+            // dependent
+            int32_t lost_in_frame = get_lost_timesamples(in_buf, in_frame_id);
+            int32_t rfi_in_frame = get_rfi_flagged_samples(in_buf, in_frame_id);
 
-            internalState& state = enabled_gated_datasets[0];
-            auto frame = visFrameView(state.buf, state.frame_id);
-            float freq_in_MHz = 800.0 - 400.0 * frame.freq_id / 1024.0;
+            // Assert that we haven't got an issue calculating the lost data
+            // This did happen when the RFI system was messing up.
+            assert(lost_in_frame >= 0);
+            assert(rfi_in_frame >= 0);
+            assert(samples_per_data_set >= (size_t)lost_in_frame);
+            assert(samples_per_data_set >= (size_t)rfi_in_frame);
 
-            long samples_in_frame =
-                samples_per_data_set - get_lost_timesamples(in_buf, in_frame_id);
+            int32_t samples_in_frame = samples_per_data_set - lost_in_frame;
 
             // Accumulate the weighted data into each dataset. At the moment this
             // doesn't really work if there are multiple frequencies in the same buffer..
             for (internalState& dset : enabled_gated_datasets) {
 
+                // TODO: CHIME specific frequency decoding
+                float freq_in_MHz = 800.0 - 400.0 * dset.frames[0].freq_id / 1024.0;
```

Comment:
There is already a function for this `freq_from_bin()` in `fpag_header_functions.h`